features/crc/mbed_crc.h

```diff
@@ -0,0 +1,102 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CRC_API_H
+#define MBED_CRC_API_H
+
+/** \addtogroup crc */
+/** @{*/
+#include <stdint.h>
+
+typedef uint64_t data_size_t;
+
+/** CRC Polynomial type
+ * 
+ * Differnet polynomials supported  
+ */
+typedef enum crc_polynomial {
+    CRC_7BIT_SD,
+    CRC_8BIT_CCITT,
+    CRC_16BIT_CCITT,
+    CRC_16BIT_SD,
+    CRC_16BIT_IBM,
+    CRC_32BIT
+} crc_polynomial_t;
+
+/** CRC width
+ * 
+ * CRC value can be 8/16/32 bit 
+ */
+typedef enum crc_width {
+    CRC_INVALID = 0,
+    CRC_7 = 7,
+    CRC_8 = 8,
+    CRC_16 = 16,
+    CRC_32 = 32,
+} crc_width_t;
+
+#define GET_DATA_SIZE(x) ((x) <= 8 ? 1 : ((x) <= 16 ? 2 : 4))
+#define TOP_BIT(x)       ((x) < 8 ? (1u << 7) : (1u << (x - 1)))
+#define DATA_MASK(x)     ((1u << x) - 1)
```

Comment:
Can these defines be moved into the C++ file?

Reply:
Yes

Reply:
If we keep this as defines (I would do local inline functions helpers to get this would have the same result and might not pollute symbols), then use prefix as for above enums (`CRC_`).

Reply:
There are two uses of `x` that are not enclosed in parentheses - could you add those?